generate-offline-map-using-android-jetpack-workmanager/src/main/java/com/esri/arcgismaps/sample/generateofflinemapusingandroidjetpackworkmanager/MainActivity.kt

```diff
@@ -0,0 +1,450 @@
+        DataBindingUtil.setContentView(this, R.layout.activity_main)
+    }
+
+    private val mapView by lazy {
+        activityMainBinding.mapView
+    }
+
+    private val takeMapOfflineButton by lazy {
+        activityMainBinding.takeMapOfflineButton
+    }
+
+    // instance of the WorkManager
+    private val workManager by lazy {
+        WorkManager.getInstance(this)
+    }
+
+    // file path to store the offline map package
+    private val offlineMapPath by lazy {
+        getExternalFilesDir(null)?.path + getString(R.string.offlineMapFile)
+    }
+
+    // shows the offline map job loading progress
+    private val progressLayout by lazy {
+        OfflineJobProgressDialogLayoutBinding.inflate(layoutInflater)
+    }
+
+    // alert dialog view for the progress layout
+    private val progressDialog by lazy {
+        createProgressDialog()
+    }
+
+    // used to uniquely identify the work request so that only one worker is active at a time
+    // also allows us to query and observe work progress
+    private val uniqueWorkName = "ArcgisMaps.Sample.OfflineMapJob.Worker"
+
+    // create a graphic overlay
+    private val graphicsOverlay = GraphicsOverlay()
+
+    // represents bounds of the downloadable area of the map
+    private val downloadArea = Graphic()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        // request notifications permission
+        requestNotificationPermission()
+
+        // authentication with an API key or named user is
+        // required to access basemaps and other location services
+        ArcGISEnvironment.apiKey = ApiKey.create(BuildConfig.API_KEY)
+        lifecycle.addObserver(mapView)
+
+        // create a portal item with the itemId of the web map
+        val portal = Portal(getString(R.string.portal_url))
+        val portalItem = PortalItem(portal, getString(R.string.item_id))
+
+        // create a symbol to show a box around the extent we want to download
+        downloadArea.symbol = SimpleLineSymbol(SimpleLineSymbolStyle.Solid, Color.red, 2F)
+        // add the graphic to the graphics overlay when it is created
+        graphicsOverlay.graphics.add(downloadArea)
+        // create and add a map with with portal item
+        val map = ArcGISMap(portalItem)
+        // apply mapview assignments
+        mapView.apply {
+            this.map = map
+            graphicsOverlays.add(graphicsOverlay)
+        }
+
+        lifecycleScope.launch {
+            map.load().onFailure {
+                // show an error and return if the map load failed
+                showMessage("Error loading map: ${it.message}")
+                return@launch
+            }
+
+            // enable the take map offline button only after the map is loaded
+            takeMapOfflineButton.isEnabled = true
+
+            // get the Control Valve layer, which is the largest layer in the
```

Comment:
Would a better wording be?
```suggestion
            // get the Control Valve layer, which is the top most layer in the
```

Reply:
Not sure if "top most" would be accurate. Seems to be the "Control Valve" layer would be 4th layer on the [web-map](https://arcgisruntime.maps.arcgis.com/home/item.html?id=acc027394bc84c2fb04d1ed317aac674). We can just drop the sentence: 
```kotlin
// get the Control Valve layer
```